modules/sor/sorV2/lib/poolsV3/gyro2CLP/gyro2CLPPool.ts

```diff
@@ -0,0 +1,353 @@
+import { Address, Hex, parseEther, parseUnits } from 'viem';
+
+import { MAX_UINT256, PoolType, SwapKind, Token, TokenAmount } from '@balancer/sdk';
+import { AddKind, RemoveKind, Vault, HookState, Gyro2CLPState } from '@balancer-labs/balancer-maths';
+import { Chain } from '@prisma/client';
+
+import { PrismaPoolAndHookWithDynamic } from '../../../../../../prisma/prisma-types';
+import { chainToChainId as chainToIdMap } from '../../../../../network/chain-id-to-chain';
+import { GyroData } from '../../../../../pool/subgraph-mapper';
+import { TokenPairData } from '../../../../../sources/contracts/v3/fetch-tokenpair-data';
+
+import { BasePoolV3 } from '../../poolsV2/basePool';
+
+import { Erc4626PoolToken, getHookState, PoolTokenWithRate, WAD } from '../../utils';
+
+import { LiquidityManagement } from '../../../../types';
+
+type GyroPoolToken = PoolTokenWithRate | Erc4626PoolToken;
+
+export class Gyro2CLPPool implements BasePoolV3 {
+    public readonly chain: Chain;
+    public readonly id: Hex;
+    public readonly address: string;
+    public readonly poolType: PoolType = PoolType.Stable;
+    public readonly swapFee: bigint;
+    public readonly tokenPairs: TokenPairData[];
+
+    public totalShares: bigint;
+    public sqrtAlpha: bigint;
+    public sqrtBeta: bigint;
+    public tokens: GyroPoolToken[];
+    public readonly hookState: HookState | undefined;
+    public readonly liquidityManagement: LiquidityManagement;
+
+    private readonly tokenMap: Map<string, GyroPoolToken>;
+
+    private vault: Vault;
+    private poolState: Gyro2CLPState;
+
+    static fromPrismaPool(
+        pool: PrismaPoolAndHookWithDynamic,
+        underlyingTokens: { address: string; decimals: number }[],
+    ): Gyro2CLPPool {
+        const poolTokens: GyroPoolToken[] = [];
+
+        if (!pool.dynamicData) throw new Error('Stable pool has no dynamic data');
```

Comment:
Should be `Gyro pool` instead of `Stable pool`

Reply:
I used the stable pool code as reference and forgot to update the error message - thanks for spotting it 🙏